clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/ApplicationCacheClient.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License
+ * for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.ecs.cache.client;
+
+import static com.netflix.spinnaker.clouddriver.ecs.EcsCloudProvider.ID;
+import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.Namespace.ECS_APPLICATIONS;
+import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.Namespace.SERVICES;
+import static java.util.stream.Collectors.toSet;
+
+import com.netflix.spinnaker.cats.cache.Cache;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.cats.cache.RelationshipCacheFilter;
+import com.netflix.spinnaker.clouddriver.ecs.cache.Keys;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsApplication;
+import com.netflix.spinnaker.clouddriver.model.Application;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class ApplicationCacheClient extends AbstractCacheClient<Application> {
+
+  @Autowired
+  public ApplicationCacheClient(Cache cacheView) {
+    super(cacheView, ECS_APPLICATIONS.toString());
+  }
+
+  @Override
+  protected Application convert(CacheData cacheData) {
+    log.debug("Translating CacheData to EcsApplication");
+    if (cacheData == null) {
+      return null;
+    }
+
+    String appName = (String) cacheData.getAttributes().get("name");
+
+    Map<String, String> attributes = new HashMap<>();
+    attributes.put("name", appName);
+
+    Map<String, Set<String>> clusterNames = new HashMap<>();
+
+    EcsApplication application = new EcsApplication(appName, attributes, clusterNames);
+
+    Set<String> services = getServiceRelationships(cacheData);
+    log.info("Found {} services for app {}", services.size(), appName);
```

Comment:
suggest we make this `debug` level instead of `info` - it's useful but produces quite a bit of noise in the logs (tested this locally myself)

Reply:
updated.